tensorboard/webapp/metrics/store/metrics_store_internal_utils.ts

```diff
@@ -296,35 +296,34 @@ export function buildOrReturnStateWithPinnedCopy(
 }
 
 /**
- * preserving the previous map, which might also contain outdated mapping due to
  * Determines which pinned cards should continue to be present in the metrics
  * state after an update to the set of experiments/tags/cards. Pinned cards are
  * removed if their corresponding card is no longer present in the state (perhaps
  * because the corresponding experiment and tag have been removed from the
- * comparison). No new pinned cards are added by this function. It generates new
+ * comparison). Pinned cards are added if they are pinned before. It generates new
```

Comment:
"pinned before" (and "present before") => just to be a little clearer about the use case, perhaps word as "previously pinned by the user"